### DISCUSSION TITLE
Several people enter data on one record...

### DISCUSSION THREAD

**Discussion by koksikus** [directus/directus#13839] [Archive]
The question is, what will happen if person 1 opens the form and works on it for an hour and in the meantime person 2 also opens it and enters his data and saves it. Which data will be saved if person 1 saves his / her data. If the data of person 1 will be preserved because they have saved later, and all data of person 2 have disappeared in this way, how to prevent this? Is it possible to implement the information that appears that someone has an open record?

**Comment by benhaynes** [comment]
I will let one of the engineers here confirm what happens now, but what I can say is we are working on real-time data with websockets that will allow for more seamless data collaboration so there are no stale/overlapping updates.

**Comment by rijkvanzanten** [comment]
As of right now, the last save operation contains the edits that are saved to the database. That being said, only the changed fields are submitted to the API, so if one user edits `title` while the other edits `author`, both changes will be saved (in order) correctly. If both users edit `title` simultaneously, whichever edit is saved last is the one that's saved to the database 👍🏻

**Reply by koksikus** [reply]
Right concept and it makes sense. Will it be possible in the future to prevent this from happening to several people at the same time on the same record in different parts of the world? It is a case in which we do not know that someone has already completed the data in this record. Or maybe it is possible now?

**Reply by koksikus** [reply]
are these the mentioned websockets? When is this option planned to be released or what will it be (what will it do)?

**Reply by rijkvanzanten** [reply]
> Will it be possible in the future to prevent this from happening to several people at the same time on the same record in different parts of the world?

Yes! That's why we brought up websockets. In order to do that efficiently, we need to have some sort of method in the API to send messages to the connected app, without the app having to poll continuously for that information. Websockets will provide that infrastructure, at which point we can start seriously looking into more live-collaboration style features, starting with preventing multiple users from editing the same fields at the same time 🙂

**Reply by koksikus** [reply]
Excellent. Now I can see on the roadmap that you are working on Real Time Data. When are you planning to release this? do you know the month of this release?

**Reply by rijkvanzanten** [reply]
[Unfortunately, we can not provide estimates on features or fixes](https://docs.directus.io/getting-started/support.html#can-you-give-an-eta-for-this-feature-fix). As of now, the only way to get an accurate timeline or release date is to [sponsor](https://docs.directus.io/getting-started/support.html#sponsored-work) this task. 👍🏻

**Reply by sxsws** [reply]
would that mean that each character change would trigger a websocket update?

**Reply by rijkvanzanten** [reply]
> would that mean that each character change would trigger a websocket update?

Don't know yet! I'm saying above that having the ability to do pub/sub communication with the API starts opening up the possibility to add more robust real-time collaboration features, but there's no set in stone specs or anything yet. I don't think we'll start with a _send every keystroke to every connected app_ as that creates a ton of overhead (and possible security no-no's).

**Reply by sxsws** [reply]
Yeah I was wondering about security too. As you say, single character updates would be impractical, also on the database, but there's a very realistic possibility that one field update could overwrite what's being written by another person if the websocket update happened only when the save button was pressed causing more conflict as one user sees their work overwritten. So there needs to be a record-level and a field-level warning stating that a field is currently being updated by another person without giving away their username.

Instant updating could be fun, e.g. a user could be in the middle of updating a record, have a bad exchange with another employee and start writing 'I f...ing hate person xyz' just to vent wtihout realising it's being broadcast everywhere which could lead to some interesting situations!
